File: desktop/src/renderer/models/Email.ts
import type { EmailAddress } from "./EmailAddress";
import type { EmailAttachment } from "./EmailAttachment";

export interface Email {
  id: string;
  thread_id: string;
  subject: string;
  from_address: EmailAddress;
  to_addresses: EmailAddress[];
  cc_addresses: EmailAddress[];
  date: string;
  body_text: string;
  body_html?: string;
  attachments: EmailAttachment[];
  labels: string[];
  snippet?: string;
  /** Boîte d'origine (gmail | outlook) pour archivage en vue "Toutes les boîtes" */
  provider?: string;
}

File: backend/core/models/Email/email.py
# pylint: disable=invalid-name
"""Modèle métier : email."""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field

from backend.core.models.Email.emailAddress import EmailAddress
from backend.core.models.Email.emailAttachment import EmailAttachment


class Email(BaseModel):
    """Représentation métier d'un email."""

    id: str
    thread_id: str
    subject: str
    from_address: EmailAddress
    to_addresses: list[EmailAddress]
    date: datetime
    body_text: str
    cc_addresses: list[EmailAddress] = Field(default_factory=list)
    bcc_addresses: list[EmailAddress] = Field(default_factory=list)
    body_html: Optional[str] = None
    attachments: list[EmailAttachment] = Field(default_factory=list)
    labels: list[str] = Field(default_factory=list)
    snippet: Optional[str] = None
    provider: Optional[str] = None  # gmail | outlook, pour archivage en vue "all"
